src/embeddings/embeddings_calculator.cc

```diff
@@ -124,31 +128,60 @@ class EmbeddingsCalculator : public CalculatorBase {
                     ov::Shape{received_batch_size},
                     strings->data()};
                 tokenizerOutputMap = tokenizer_session->infer(tokenizerInputMap);
-            } else {
-                // TODO: input already tokenized
-                return absl::InvalidArgumentError("not implemented");
-            }
-            ::InferenceInput embeddingsInputMap;
-            // Check if tokenizer produced at least the number of outputs as there are inputs in embedding model
-            RET_CHECK(tokenizerOutputMap.size() >= embeddingsInputNames.size());
-            for (const auto& embeddingsInputName : embeddingsInputNames) {
-                auto it = tokenizerOutputMap.find(embeddingsInputName);
-                RET_CHECK(it != tokenizerOutputMap.end());
-                SPDLOG_LOGGER_DEBUG(embeddings_calculator_logger, "Embedding model input {} is connected with matching tokenizer output", embeddingsInputName);
-                embeddingsInputMap[embeddingsInputName] = it->second;
+                // Check if tokenizer produced at least the number of outputs as there are inputs in embedding model
+                RET_CHECK(tokenizerOutputMap.size() >= embeddingsInputNames.size());
+                for (const auto& embeddingsInputName : embeddingsInputNames) {
+                    auto it = tokenizerOutputMap.find(embeddingsInputName);
+                    RET_CHECK(tokenizerOutputMap[EMBEDDINGS_MODEL_INPUT_IDS_NAME].get_shape().size() == 2);
+                    size_t input_ids_size = tokenizerOutputMap[EMBEDDINGS_MODEL_INPUT_IDS_NAME].get_shape()[1];
+                    if (input_ids_size > max_context_length) {
+                        SPDLOG_LOGGER_DEBUG(embeddings_calculator_logger, "Input size {} exceeds max_context_length {}", input_ids_size, max_context_length);
+                        return absl::InvalidArgumentError(absl::StrCat("Input length ", input_ids_size, " longer than allowed ", max_context_length));
+                    }
+                    RET_CHECK(it != tokenizerOutputMap.end());
+                    SPDLOG_LOGGER_DEBUG(embeddings_calculator_logger, "Embedding model input {} is connected with matching tokenizer output", embeddingsInputName);
+                    embeddingsInputMap[embeddingsInputName] = it->second;
+                }
+            } else if (auto ints = std::get_if<std::vector<std::vector<int>>>(&input)) {
+                size_t max_b = 0;
+                std::vector<int> input_ids;
+                std::vector<int> attention_mask;
```

Comment:
`int` size is dependend on architecture, and we need to be sure that it is always 8 bytes long. please use the same precision as model type (int64_t?)

Reply:
As you can see @dtrawins embedding endpoint will also support only 1 precision for tokens - int64. We could templatize this for both cases, but since we see no model with int32, uint32, uint64 precisions it should be low priority.

@dtrawins